source/log_analyzer/log_analyzer.py

```diff
@@ -0,0 +1,182 @@
+        self.api = api
+        self.sub_request_id = sub_request_id
+        self.worker = worker
+        self.start = int(start)
+        self.end = int(end)
+
+
+class JormunTask:
+    def __init__(self, task_name, sub_request_id, start, end):
+        self.task_name = task_name
+        self.sub_request_id = sub_request_id
+        self.start = int(start)
+        self.end = int(end)
+
+
+class Request:
+    def __init__(self, request_id, scenario):
+        self.request_id = request_id
+        self.scenario = scenario
+        self.kraken_tasks_by_worker = {}  # dict worker -> list of tasks
+        self.jormun_tasks = []
+        self.url = ""
+
+    def add_kraken_task(self, api, sub_request_id, worker, start, end):
+        worker_tasks = self.kraken_tasks_by_worker.get(worker, [])
+        worker_tasks.append(KrakenTask(api, sub_request_id, worker, start, end))
+        self.kraken_tasks_by_worker[worker] = worker_tasks
+
+    def add_jormun_task(self, task_name, sub_request_id, start, end):
+        self.jormun_tasks.append(JormunTask(task_name, sub_request_id, start, end))
+
+    def set_url(self, url):
+        self.url = url
+
+    def myprint(self):
+        print("Request : " + self.request_id + " scenario : " + self.scenario + " url : " + self.url)
+        for worker, tasks in self.kraken_tasks_by_worker.items():
+            print("   Worker : " + worker)
+            for task in tasks:
+                print(
+                    "      Api : " + task.api + " sub_request : " + task.sub_request_id + " start : ",
+                    task.start,
+                    " end : ",
+                    task.end,
+                )
+        print("   Jormun : ")
+        for task in self.jormun_tasks:
+            print(
+                "      Task : " + task.task_name + " sub_request : " + task.sub_request_id + " start : ",
+                task.start,
+                " end : ",
+                task.end,
+            )
+
+    def create_gantt(self, output_dir):
+        # https://plotly.com/python/gantt/
+        import plotly.figure_factory
+
+        sorted_jormun_tasks = sorted(self.jormun_tasks, key=lambda task: (task.start, -task.end))
+        df = [
+            dict(
+                Task="J{}#{}".format(task.sub_request_id, task.task_name),
+                Start=task.start,
+                Finish=task.end,
+                SubRequest="J{}#{}".format(task.sub_request_id, task.task_name),
+            )
+            for task in sorted_jormun_tasks
+        ]
+        for worker, tasks in self.kraken_tasks_by_worker.items():
+            for task in tasks:
+                df.append(
+                    dict(
+                        Task="Kraken_{}".format(task.worker),
+                        Start=task.start,
+                        Finish=task.end,
+                        SubRequest="K{}".format(task.sub_request_id),
+                    )
+                )
```

Comment:
just for the sake of pythoness and a bit of fun, this could be refacto a little:

```py
def make_df(task):
   return dict(   Task="Kraken_{}".format(task.worker),
                        Start=task.start,
                        Finish=task.end,
                        SubRequest="K{}".format(task.sub_request_id))

f = [ make_df(task) for task in sorted_jormun_tasks ]
f.extends([ make_df(t) for tasks in self.kraken_tasks_by_worker.values() for t in tasks ])
```